### PR TITLE
Trim AUTO_INCREMENT values from MySQL schema dumps

### LINKED ISSUE #56: schema dump includes `AUTO INCREMENT`
Hi,

When I `dbmate migrate` the schema dump includes the current `AUTO INCREMENT` of each table, which is bad since if I want to import the schema as is (without going through all the migrations) to a server or another computer it will start the row count from where my local table ends.

Thanks

### PR DESCRIPTION
**Pull request by shames0**
closes #56

### CODE REVIEW

File: pkg/driver/mysql/mysql.go [hunk 1]
```diff
@@ -192,7 +193,17 @@ func (drv *Driver) DumpSchema(db *sql.DB) ([]byte, error) {
...
-	return dbutil.TrimLeadingSQLComments(schema)
+	schema, err = dbutil.TrimLeadingSQLComments(schema)
+	if err != nil {
+		return nil, err
+	}
+	return TrimAutoincrementValues(schema), nil
+}
+
+// TrimAutoincrementValues removes AUTO_INCREMENT values from MySQL schema dumps
+func TrimAutoincrementValues(data []byte) []byte {
```
Comment (amacneil) [inline pkg/driver/mysql/mysql.go:204]: I think this can start with a lowercase letter since it does not need to be exported?

**Comment by amacneil** [comment]
Looks like a great start. Can you also add some tests for it?

**Comment by shames0** [comment]
Thanks for the feedback, I'll look at adding tests later this evening.

**Comment by shames0** [comment]
Tests added, ready for any other feedback you might have. This is my first time writing GO... so feel free to nudge me toward best practices.

File: pkg/driver/mysql/mysql_test.go [hunk 2]
```diff
@@ -183,6 +183,35 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+	var tblName, tblCreate string
+	err = db.QueryRow(`show create table foo_table`).Scan(&tblName, &tblCreate)
+	require.NoError(t, err)
+
+	require.Contains(t, tblCreate, "AUTO_INCREMENT=2")
+
+	// AUTO_INCREMENT should not appear in the dump
+	schema, err := drv.DumpSchema(db)
+	require.NoError(t, err)
+	require.NotContains(t, string(schema), "AUTO_INCREMENT=2")
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:212]:
```suggestion
	require.NotContains(t, string(schema), "AUTO_INCREMENT")
```

File: pkg/driver/mysql/mysql_test.go [hunk 3]
```diff
@@ -183,6 +183,35 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+	// create a record
+	_, err = db.Exec(`insert into foo_table values ()`)
+	require.NoError(t, err)
+
+	// AUTO_INCREMENT should appear on the table definition
+	var tblName, tblCreate string
+	err = db.QueryRow(`show create table foo_table`).Scan(&tblName, &tblCreate)
+	require.NoError(t, err)
+
+	require.Contains(t, tblCreate, "AUTO_INCREMENT=2")
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:207]:
```suggestion
	require.Contains(t, tblCreate, "AUTO_INCREMENT")
```

File: pkg/driver/mysql/mysql_test.go [hunk 4]
```diff
@@ -183,6 +183,35 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+
+	// create a record
+	_, err = db.Exec(`insert into foo_table values ()`)
+	require.NoError(t, err)
+
+	// AUTO_INCREMENT should appear on the table definition
+	var tblName, tblCreate string
+	err = db.QueryRow(`show create table foo_table`).Scan(&tblName, &tblCreate)
+	require.NoError(t, err)
+
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:206]:
```suggestion
	require.NoError(t, err)
```

File: pkg/driver/mysql/mysql_test.go [hunk 5]
```diff
@@ -183,6 +183,35 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+	require.Contains(t, tblCreate, "AUTO_INCREMENT")
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:207]:
```suggestion
	require.NoError(t, err)
	require.Contains(t, tblCreate, "AUTO_INCREMENT")
```

**Comment by amacneil** [comment]
thank you!

File: pkg/driver/mysql/mysql_test.go [hunk 6]
```diff
@@ -183,6 +183,34 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+
+	// create a record
+	_, err = db.Exec(`insert into foo_table values ()`)
+	require.NoError(t, err)
+
+	// AUTO_INCREMENT should appear on the table definition
+	var tblName, tblCreate string
+	err = db.QueryRow(`show create table foo_table`).Scan(&tblName, &tblCreate)
+	require.NoError(t, err)
+	require.Contains(t, tblCreate, "AUTO_INCREMENT")
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:206]:
```suggestion
	require.Contains(t, tblCreate, "AUTO_INCREMENT=")
```

File: pkg/driver/mysql/mysql_test.go [hunk 7]
```diff
@@ -183,6 +183,34 @@ func TestMySQLDumpSchema(t *testing.T) {
...
+
+	// AUTO_INCREMENT should not appear in the dump
+	schema, err := drv.DumpSchema(db)
+	require.NoError(t, err)
+	require.NotContains(t, string(schema), "AUTO_INCREMENT")
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:211]:
```suggestion
	require.NotContains(t, string(schema), "AUTO_INCREMENT=")
```

File: pkg/driver/mysql/mysql_test.go [hunk 8]
```diff
@@ -183,6 +183,34 @@ func TestMySQLDumpSchema(t *testing.T) {
...
 
+func TestMySQLDumpSchemaContainsNoAutoIncrement(t *testing.T) {
+	drv := testMySQLDriver(t)
+
+	db := prepTestMySQLDB(t)
+	defer dbutil.MustClose(db)
+	err := drv.CreateMigrationsTable(db)
+	require.NoError(t, err)
+
+	// create table with AI column
```
Comment (amacneil) [inline pkg/driver/mysql/mysql_test.go:194]:
```suggestion
	// create table with AUTO_INCREMENT column
```

**Pull request merged**